superset/views/core.py

```diff
@@ -758,8 +758,13 @@ class R(BaseSupersetView):
     @expose('/<url_id>')
     def index(self, url_id):
         url = db.session.query(models.Url).filter_by(id=url_id).first()
-        if url:
-            return redirect('/' + url.url)
+        if url and url.url:
+            explore_url = '//superset/explore/?'
+            if url.url.startswith(explore_url):
+                explore_url += 'r={}'.format(url_id)
+                return redirect('/' + explore_url)
```

Comment:
Doesn't that make for a triple slash here? I mean as in '///superset/explore/?r=423'

Reply:
yes, we generated a str like '///superset/explore/?r=423'. 
I copied this extra '/' from existed code...and if i removed this extra '/', it will redirect to wrong path. 😳

Reply:
@williaster from my understanding this redirect is only for parent pages and not ajax call. Maybe the browser is cool with 3 slashes but the ajax calls are not(?).

I'd vote for taking out the slash and testing it works well for all types of urls and move forward without it.